gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -447,16 +573,30 @@ private ManagedChannel createSingleChannel() throws IOException {
       builder.keepAliveTime(DIRECT_PATH_KEEP_ALIVE_TIME_SECONDS, TimeUnit.SECONDS);
       builder.keepAliveTimeout(DIRECT_PATH_KEEP_ALIVE_TIMEOUT_SECONDS, TimeUnit.SECONDS);
     } else {
+      // Try and create credentials via DCA. See https://google.aip.dev/auth/4114.
       ChannelCredentials channelCredentials;
       try {
         channelCredentials = createMtlsChannelCredentials();
       } catch (GeneralSecurityException e) {
         throw new IOException(e);
       }
       if (channelCredentials != null) {
-        builder = Grpc.newChannelBuilder(endpoint, channelCredentials);
+        // Create the channel using channel credentials created via DCA.
+        builder = Grpc.newChannelBuilder(endpointContext.resolvedEndpoint(), channelCredentials);
```

Comment:
There is a concern that `endpointContext.resolvedEndpoint()` may be different from `endpoint` especially when TransportChannelProvider.withEndpoint(...) is used directly by the customers.

I'll will try confirm this behavior work as intended, otherwise we may need to revert the change that is using `endpointContext.resolvedEndpoint()`.

Reply:
Ack. (just to make it easier to revert if needed) the change is split between https://github.com/googleapis/sdk-platform-java/pull/3326#discussion_r1831368644 and https://github.com/googleapis/sdk-platform-java/pull/3326/commits/4ee2ee91154b83def30ae44fce7424674f616f80. 

Out of curiosity:
My understanding that these are effectively the same, as endpoint is always initialized to the resolved endpoint before being passed to the transport channel provider. In ClientContext for example, the `endpoint` is set to `endpointContext.resolvedEndpoint()`:
https://github.com/googleapis/sdk-platform-java/blob/01e742de49e151efbd903808a859f595f99bc8de/gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java#L180

Is this accurate?

Reply:
You're right that they *should* be the same. I am just going to run some tests to confirm this behavior and see if anything would break that assumption.

Reply:
Hmm, I think we should revert the part inside `createSingleChannel()` that use endpointContext's resolvedEndpoint. For most use cases (especially with the default client library behavior), the endpoint would be the endpointContext's value and it will work as we expect.

However, I believe we have some use existing cases for FixedTransportChannelProvider that don't pass EndpointContext and this will fail and error out:
```
    InstantiatingGrpcChannelProvider grpcChannelProvider = InstantiatingGrpcChannelProvider.newBuilder()
            .setEndpoint("testEndpoint.com:443").setHeaderProvider(FixedHeaderProvider.create("hello", "world")).build();
    TransportChannel transportChannel = grpcChannelProvider.getTransportChannel();
    FixedTransportChannelProvider fixedTransportChannelProvider = FixedTransportChannelProvider.create(transportChannel);
```

EndpointContext was not passed to grpcChannelProvider and will result in a NPE.
---
I think the right move is to just revert it inside this method. In a separate PR, I can add comments to explain why both exists and add the necessary @ObsoleteApi annotations.

Reply:
Thank you for investigating Lawrence and pointing out this will cause breakage when using the FixedTransportChannelProvider!

In https://github.com/googleapis/sdk-platform-java/pull/3326/commits/c96c76092c69755007878ace8a44acc56e8a59ec I reverted usage of `endpointContext.resolvedEndpoint()` in all places (in the grpc and http channel providers), not just the `createSingleChannel()`.  In grpc: outside `createSingleChannel()` there is only one other usage, in `canUseDirectPathWithUniverseDomain`, so I figured reverting everywhere would be cleaner for now. In http: there was only one usage of `endpointContext.resolvedEndpoint()`, I think it would be safe to keep, but reverted it also. Let me know if we want to bring any of these usage of `endpointContext.resolvedEndpoint()` in grpc or http back in this PR, or if you want to do it when you obsolete the api's in a separate PR.

`mvn test` is green.